src/Composer/Util/Zip.php

```diff
@@ -105,6 +112,10 @@ private static function locateFile(\ZipArchive $zip, $filename)
             return $index;
         }
 
+        if ($index && $foundFileCount > 1) {
+            throw new MultipleComposerJsonException('Multiple composer.json files were found.');
+        }
+
         // no composer.json found either at the top level or within the topmost directory
         return false;
```

Comment:
Here should throw with a message saying what the comment says, because otherwise currently if no composer.json is in at all you get `false` but without any feedback. I think this function should ideally always return a file index or throw with a correct exception message.